docs/spec/mint/begin_block.md

```diff
@@ -26,3 +26,26 @@ NextInflation(params Params, bondedRatio sdk.Dec) (inflation sdk.Dec) {
 	}
 
 	return inflation
+```
+
+## NextAnnualProvisions
+
+Calculate the annual provisions based on current total supply and inflation
+rate. This parameter is calculated once per block. 
+
+```
+NextAnnualProvisions(params Params, totalSupply sdk.Dec) (provisions sdk.Int) {
+	provisionsDec = Inflation * totalSupply
+	return provisionsDec.Truncate()
+```
+
+## BlockProvision
+
+Calculate the provisions generated for each block based on current 
+annual provisions 
+
+```
+BlockProvision(params Params) sdk.Coin {
+	provisionAmt = AnnualProvisions/ params.BlocksPerYear
```

Comment:
(can we truncate after this calculation instead)

Reply:
++

Reply:
see above